src/index.js

```diff
@@ -9,11 +9,17 @@ const supportMultiple = (typeof document !== 'undefined' && document && document
 
 class Dropzone extends React.Component {
   static renderChildren(children, isDragActive, isDragReject) {
+    if (!children) {
+      return children;
+    }
     let renderedChildren = children;
+    const props = { isDragActive, isDragReject };
     if (typeof children === 'function') {
-      renderedChildren = children(this.state);
+      renderedChildren = children(props);
     }
-    return React.cloneElement(renderedChildren, { isDragActive, isDragReject });
+    // a warning gets printed because we are giving unkown props to some children (<p>some content</p> for instance)
+    renderedChildren = React.cloneElement(renderedChildren, props);
```

Comment:
In my opinion, doing this is bad. It may break some people's code (if they had components with `isDragActive` prop for instance). I think we should remove this step.